src/lib/os-release.ts

```diff
@@ -1,49 +1,60 @@
-import * as Bluebird from 'bluebird';
-import * as fs from 'fs';
 import * as _ from 'lodash';
+import fs = require('mz/fs');
 
+import { InternalInconsistencyError } from './errors';
 import log from './supervisor-console';
 
-// FIXME: Don't use synchronous file reading and change call sites to support a promise
-function getOSReleaseField(path: string, field: string): string | undefined {
-	try {
-		const releaseData = fs.readFileSync(path, 'utf-8');
-		const lines = releaseData.split('\n');
-		const releaseItems: { [field: string]: string } = {};
-
-		for (const line of lines) {
-			const [key, value] = line.split('=');
-			releaseItems[_.trim(key)] = _.trim(value);
+// Retrieve the data for the OS once only per path
+const getOSReleaseData = _.memoize(
+	async (path: string): Promise<Dictionary<string>> => {
+		const releaseItems: Dictionary<string> = {};
+		try {
+			const releaseData = await fs.readFile(path, 'utf-8');
+			const lines = releaseData.split('\n');
+
+			for (const line of lines) {
+				const [key, ...values] = line.split('=');
```

Comment:
```suggestion
				const [key, quatedValue] = line.split('=', 2);
```

This would remove the necessity to join values 2 lines below:
```
const value = _.trim(quatedValue.replace(...))
```